helix-term/src/commands/typed.rs

```diff
@@ -2408,6 +2408,57 @@ fn redraw(
     Ok(())
 }
 
+fn move_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    ensure!(args.len() == 1, format!(":move takes one argument"));
+
+    let new_path = args.first().unwrap().to_string();
+    let doc = doc!(cx.editor);
+    let old_path = doc
+        .path()
+        .ok_or_else(|| anyhow!("Scratch buffer cannot be moved. Use :write instead"))?
+        .to_string_lossy()
+        .to_string();
+
+    let lsp_reply = doc.language_servers().find_map(|lsp| {
+        if let Some(future) = lsp.prepare_file_rename(old_path.clone(), new_path.clone()) {
+            match helix_lsp::block_on(future) {
+                Ok(edit) => return Some(edit),
+                Err(e) => {
+                    log::error!("LSP willRename request failed: {:?}", e);
+                }
+            }
+        };
+        None
+    });
+
+    if let Some(edit) = lsp_reply {
+        if let Err(e) = apply_workspace_edit(cx.editor, helix_lsp::OffsetEncoding::Utf8, &edit) {
```

Comment:
hardcoding the offset encodign to utf8 is always incorrect. You need to use the correct offset encoding for the server

Reply:
updated :)